src/librustc_trans/abi.rs

```diff
@@ -283,10 +287,19 @@ impl<'tcx> LayoutExt<'tcx> for TyLayout<'tcx> {
                 })
             }
 
-            Layout::Vector { .. } => {
+            // LLVM has a separate type for 64-bit SIMD vectors on X86 called
+            // `x86_mmx` which is needed for some SIMD operations. As a bit of a
+            // hack (all SIMD definitions are super unstable anyway) we
+            // recognize any one-element SIMD vector as "this should be an
+            // x86_mmx" type. In general there shouldn't be a need for other
+            // one-element SIMD vectors, so it's assumed this won't clash with
+            // much else.
+            Layout::Vector { count, .. } => {
+                let size = self.size(ccx);
+                let x86_mmx = count == 1;
```

Comment:
Can you also check `size.bits() == 64` and maybe the architecture? Otherwise this is a neat hack!